docs: Move TopK example code to extending-operators documentation

gowtham1412-p opened the pull request:
## Which issue does this PR close?

Closes #15774

## Rationale for this change

Moving the TopK example code from the test file to user-facing documentation makes it more discoverable for users learning how to create custom operators in DataFusion.

## What changes are included in this PR?

- Added TopK operator example to `extending-operators.md` documentation
- Includes logical plan node, optimizer rule, physical planner, and execution plan examples
- Added usage instructions for implementing custom TopK queries

## Are these changes tested?

Documentation-only change, no functional code changes.

## Are there any user-facing changes?

Yes - users can now find the TopK custom operator example in the official documentation instead of having to browse test files.
```

### **2. Add a Comment on Issue #15774**

Go to https://github.com/apache/datafusion/issues/15774 and comment:
```
I've opened PR #18372 to address this issue. Please review at your earliest convenience.

Jefffrey reviewed (commented):
Is completely replacing the existing contents of `extending-operators.md` page an intended change?

gowtham1412-p commented:
@Jefffrey I've updated the PR to preserve the existing µWheel documentation. The TopK example is now added as a second example rather than replacing the original content. Please review when you have the convienent time

alamb commented on docs/source/library-user-guide/extending-operators.md:
```diff
@@ -19,22 +19,21 @@
   plan: LogicalPlan,
   _config: &dyn OptimizerConfig,
-) -> Result<Transformed<LogicalPlan>> {
+) -> Result<Transformed> {
```
I am pretty sure this is an incorrect change --

Jefffrey commented:
> please review it once.

Do you have any thoughts on the comment I left above?

Jefffrey commented on docs/source/library-user-guide/extending-operators.md:
````diff
@@ -43,11 +42,9 @@ fn rewrite(
-```rust,ignore
 // Converts a uwheel aggregate result to a TableScan with a MemTable as source
-fn agg_to_table_scan(result: f64, schema: SchemaRef) -> Result<LogicalPlan> {
+fn agg_to_table_scan(result: f64, schema: SchemaRef) -> Result {
````
I'm very confused why these changes are being made, especially considering it makes the example invalid?

Jefffrey reviewed (commented):
The general layout is looking better, but there's details like syntax being incorrect which I'm quite confused by considering the example code already exists in our codebase 🤔
